libsel4vm/arch_include/arm/sel4vm/arch/guest_vm_arch.h

```diff
@@ -15,15 +15,27 @@
 #include <sel4vm/guest_vm.h>
 
 typedef struct fault fault_t;
+typedef struct vm vm_t;
 typedef struct vm_vcpu vm_vcpu_t;
 
+typedef seL4_CPtr(*get_smc_cap_callback_fn)(seL4_Word smc_call);
```

Comment:
> (Here you can have that space after seL4_Cptr.)

The styler still updated this one. I think it is some weird edge case.

> However, this callback is not needed because the user can easily call smc_forward() themselves in their smc_handle_callback function. One of the reasons for me proposing this change was the awkward return cap API. Using smc_handle_callback also for forwarding gives the user a chance to decide between faulting and returning an error if the cap can't be found.

So CAmkES commonly generates functions for its components so it can get its caps. The library doesn't know what that function is, so the default handler needs some way to get the cap. I am not sure of a better wat to do this if you are using the default and don't define your own `smc_handle_callback`.

Reply:
ahhhh, sorry. I have misunderstood your proposal the whole time. That makes sense to me now I will try that out.

Reply:
I had the feeling that was the case, but other than repeating the same thing in different ways I didn't know how to resolve it. Anyway, we got there in the end.